cmd/kube-controller-manager/app/options/nodelifecyclecontroller.go

```diff
@@ -44,7 +44,8 @@ func (o *NodeLifecycleControllerOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.Float32Var(&o.SecondaryNodeEvictionRate, "secondary-node-eviction-rate", 0.01, "Number of nodes per second on which pods are deleted in case of node failure when a zone is unhealthy (see --unhealthy-zone-threshold for definition of healthy/unhealthy). Zone refers to entire cluster in non-multizone clusters. This value is implicitly overridden to 0 if the cluster size is smaller than --large-cluster-size-threshold.")
 	fs.Int32Var(&o.LargeClusterSizeThreshold, "large-cluster-size-threshold", 50, "Number of nodes from which NodeController treats the cluster as large for the eviction logic purposes. --secondary-node-eviction-rate is implicitly overridden to 0 for clusters this size or smaller.")
 	fs.Float32Var(&o.UnhealthyZoneThreshold, "unhealthy-zone-threshold", 0.55, "Fraction of Nodes in a zone which needs to be not Ready (minimum 3) for zone to be treated as unhealthy. ")
-	fs.BoolVar(&o.EnableTaintManager, "enable-taint-manager", o.EnableTaintManager, "WARNING: Beta feature. If set to true enables NoExecute Taints and will evict all not-tolerating Pod running on Nodes tainted with this kind of Taints.")
+	fs.BoolVar(&o.EnableTaintManager, "enable-taint-manager", o.EnableTaintManager, "If set to true enables NoExecute Taints and will evict all not-tolerating Pod running on Nodes tainted with this kind of Taints.")
```

Comment:
Is this flag functional today? What if someone doesn't want to run this?

Reply:
It is functional. If you set it to false, the older codepath runs. But we can't keep maintaining that path.

Reply:
@damemi I think you graduated this to GA. Was this CLI flag left on purpose?

Reply:
@alculquicondor no, I think it was just an oversight, didn't realize there was a command-line flag along with the feature gates so that's probably why it got overlooked.

Reply:
I have the same question as @lavalamp, this patch is based on an assumption that all people need to run taint manager, is that true?

> If you set it to false, the older codepath runs. But we can't keep maintaining that path.

In addition, I'm thinking what's the benefit we can get from the deprecation work. Seems like it helps reduce the maintenance effort, right?
Can you give a link to the `older codepath`?

Reply:
> this patch is based on an assumption that all people need to run taint manager

The assumption is that the newer codepath does everything that the older codepath does, so there is no need to keep maintaining both.

> In addition, I'm thinking what's the benefit we can get from the deprecation work. Seems like it helps reduce the maintenance effort, right?

Yes. This deprecation is motivated by #110959, where we were wondering if we should add new logic to this old codepath. And it doesn't really make sense to do so.

> Can you give a link to the older codepath?

It's hard to visualize, but it's embedded in https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/nodelifecycle/node_lifecycle_controller.go
when `runTaintManager=false`